syntax = "proto3";


package spiderrpc;


service SpiderRpc {
    rpc SubmitRssFeed(RssFeed) returns (SubmitResult) {}
    rpc CrawlArticles(SpiderList) returns (CrawlTaskResult) {}
}

message RssFeed {
    string url = 1;
    string category = 2;
    string item_content_xpath = 3;
    repeated string removed_xpath_nodes = 4;
    string css = 5;
}

message SubmitResult {
    bool error = 1;
    string message = 2;
}

message SpiderList {
    repeated string spider = 1;
}

message CrawlTaskResult {
    bool isrunning = 1;
}
